internal/statemachine/state_machine.go

```diff
@@ -129,6 +129,9 @@ type StateMachine struct {
 	// names of images for each volume
 	VolumeNames map[string]string
 
+	// name of the "main volume"
+	MainVolumeName string
```

Comment:
I am possibly missing something here so I apologize if this is obvious, but how is the `MainVolumeName` field used? I see it being set, but... I don't actually see it being consumed anywhere? What did I miss?

Reply:
No you are completely right, **this is currently unused**. This was done on purpose. When working on this and talking with Alfonso, we came to the conclusion that the "main volume" should be the one with the `system-seed` role. And calling `gadget.Validate()` (which we did not do before) ensure there is a single structure with this role.

BUT, considering this one to be the "main one" is not really standard and I am not 100% sure it is the currently the case for our users. See the comment here https://github.com/canonical/ubuntu-image/pull/217/files#diff-c31b17e50caf5e415f8b656ff1a4d6ca2bcaf9460e272407bebbe3279090d5f9R405

So my idea was to make it explicit in the code for now that we decide what is the main volume, and then later, when we are sure that the new validations this PR introduce are not breaking anything (or that users fixed their gadgets) we can move to the next step and enforce it. So for now we keep the fact that the last volume is the "main one".

Reply:
Okay, thank you for the explanation!